src/ds/ds-device-common.cpp

```diff
@@ -208,6 +208,12 @@ namespace librealsense
                 if (index >= offset + size)
                     break;
                 int packet_size = std::min((int)(HW_MONITOR_COMMAND_SIZE - (i % HW_MONITOR_COMMAND_SIZE)), (int)(ds::FLASH_SECTOR_SIZE - i));
+
+                // This is a WORKAROUND to support MIPI camera D457.
+                // If the packet size is more than 128 bytes, FLASH_WRITE is not working as expected.
+                // Revert this line after fixing the issue.
+                packet_size = std::min(packet_size, 128);
```

Comment:
This is for all D400 devices? not MIPI only?

Reply:
Right now it is for all. Maybe for D500 devices as well.

Reply:
Updated it to only apply this workaround for D457.